packages/core/src/domain/telemetry/rawTelemetryEvent.types.ts

```diff
@@ -0,0 +1,34 @@
+import type { Context } from '../../tools/context'
+
+export const enum TelemetryType {
+  log = 'log',
+  configuration = 'configuration',
+}
+
+export const enum StatusType {
+  debug = 'debug',
+  error = 'error',
+}
+
+export type RawTelemetryEvent = RawDebugTelemetryEvent | RawErrorTelemetryEvent | RawConfigurationTelemetryEvent
+
+export interface RawDebugTelemetryEvent extends Context {
+  message: string
+  status: StatusType.debug
+  type: TelemetryType.log
+  error?: {
+    kind?: string
+    stack: string
+  }
+}
+
+export interface RawErrorTelemetryEvent extends Context {
+  message: string
+  status: StatusType.error
+  type: TelemetryType.log
+}
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**❗ warning:** ‏that does not seem right 😕

<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Is there a reason to declare Raw types structure? could we use directly the `telemetry` field from the format?
```ts
export type RawDebugTelemetryEvent = TelemetryDebugEvent['telemetry']
```

Reply:
The main reason is that Event types have `[k: string]: unknown` at every depth. Meaning that unknown attributes are allowed. Ex:
```typescript
  addTelemetry(
    assign(
      {
        type: TelemetryType.log as const,
        message,
        status: StatusType.debug as const,
        foo: 'bar', // not detected by the linter
      },
      context
    )
  )`
Also, we have to be backward compat in the format so field like `telemetry.type` is nullable where it should not.
At least if we where to remove `[k: string]: unknown` I would be more comfortable. Do you know why is it set in the first place?

Reply:
My example is not the best since debug and error telemetry can accept other field. However configuration telemetry should not.

Reply:
I made the change

Reply:
> At least if we where to remove [k: string]: unknown I would be more comfortable. Do you know why is it set in the first place?

I think it is because the types are generated to allow forward compatibility.
I am fine with using RawTypes but they have to match the telemetry types (cf previous inversion between Error and Debug) and we should not use `any`.